src/java/org/apache/cassandra/repair/consistent/LocalSessions.java

```diff
@@ -637,10 +642,13 @@ public LocalSession getSession(TimeUUID sessionID)
     }
 
     @VisibleForTesting
-    synchronized void putSessionUnsafe(LocalSession session)
+    synchronized boolean putSessionUnsafe(LocalSession session)
     {
+        if (sessions.containsKey(session.sessionID))
```

Comment:
this actually makes it safe... but does have a big change in semantics... for retry logic we need ability to dedup, so blind overriding can't be done... I could always rename this to `putSession`....